packages/playwright-test/src/index.ts

```diff
@@ -617,9 +618,15 @@ class ArtifactsRecorder {
         await Promise.all(context.pages().map(async page => {
           if ((page as any)[this._screenshottedSymbol])
             return;
-          // Pass caret=initial to avoid any evaluations that might slow down the screenshot
-          // and let the page modify itself from the problematic state it had at the moment of failure.
-          await page.screenshot({ ...this._screenshotOptions, timeout: 5000, path: this._addScreenshotAttachment(), caret: 'initial' }).catch(() => {});
+          try {
+            const screenshotPath = this._createScreenshotAttachmentPath();
+            // Pass caret=initial to avoid any evaluations that might slow down the screenshot
+            // and let the page modify itself from the problematic state it had at the moment of failure.
+            await page.screenshot({ ...this._screenshotOptions, timeout: 5000, path: screenshotPath, caret: 'initial' });
+            this._testInfo.attachments.push({ name: 'screenshot', path: screenshotPath, contentType: 'image/png' });
```

Comment:
While I appreciate the strictless here, I think the actual bug is in the tracing processing that assumes that all the attachments always exist. I think we should fix it there first.